src/fontra/views/editor/visualization-layer-definitions.js

```diff
@@ -851,6 +981,117 @@ registerVisualizationLayerDefinition({
   },
 });
 
+registerVisualizationLayerDefinition({
+  identifier: "fontra.selected.guidelines",
+  name: "Selected guidelines",
+  selectionMode: "editing",
+  zIndex: 500,
+  screenParameters: {
+    smoothSize: 8,
+    strokeWidth: 1,
+    hoverStrokeOffset: 4,
+    underlayOffset: 2,
+    iconSize: 12,
+  },
+  colors: {
+    hoveredColorIcon: "#555",
+    hoveredColor: "#BBB",
+    selectedColor: "#000",
+    underColor: "#FFFA",
+  },
+  colorsDarkMode: {
+    hoveredColorIcon: "#BBB",
+    hoveredColor: "#BBB",
+    selectedColor: "#FFF",
+    underColor: "#0008",
+  },
+  draw: (context, positionedGlyph, parameters, model, controller) => {
+    const glyph = positionedGlyph.glyph;
+    const smoothSize = parameters.smoothSize;
+
+    const {
+      guideline: hoveredGuidelineIndices,
+      fontGuideline: hoveredFontGuidelineIndices,
+    } = parseSelection(model.hoverSelection);
+    const {
+      guideline: selectedGuidelineIndices,
+      fontGuideline: selectedFontGuidelineIndices,
+    } = parseSelection(model.selection);
+
+    // TODO: Font Guidelines
+
+    // Under layer
+    context.fillStyle = parameters.underColor;
+    for (const i of selectedGuidelineIndices || []) {
+      const guideline = glyph.guidelines[i];
+      if (!guideline) {
+        continue;
+      }
+      if (guideline.locked) {
+        _drawLockIcon(
+          context,
+          parameters,
+          guideline.x - parameters.iconSize / 2,
+          guideline.y + parameters.iconSize / 2
+        );
+      } else {
+        fillRoundNode(context, guideline, smoothSize + parameters.underlayOffset);
+      }
+    }
+
+    // Selected guideline
+    context.fillStyle = parameters.selectedColor;
+    for (const i of selectedGuidelineIndices || []) {
+      const guideline = glyph.guidelines[i];
+      if (!guideline) {
+        continue;
+      }
+      if (guideline.locked) {
+        context.strokeStyle = parameters.selectedColor;
+        _drawLockIcon(
+          context,
+          parameters,
+          guideline.x - parameters.iconSize / 2,
+          guideline.y + parameters.iconSize / 2
+        );
+      } else {
+        fillRoundNode(context, guideline, smoothSize);
+      }
+    }
+
+    // // Hovered guideline
+    context.strokeStyle = parameters.hoveredColor;
+    context.lineWidth = parameters.strokeWidth;
+    for (const i of hoveredGuidelineIndices || []) {
+      const guideline = glyph.guidelines[i];
+      if (!guideline) {
+        continue;
+      }
+      if (guideline.locked) {
+        context.strokeStyle = parameters.hoveredColorIcon;
+        _drawLockIcon(
+          context,
+          parameters,
+          guideline.x - parameters.iconSize / 2,
+          guideline.y + parameters.iconSize / 2
+        );
+      } else {
+        strokeRoundNode(context, guideline, smoothSize + parameters.hoverStrokeOffset);
+      }
+    }
+  },
+});
+
+function _drawLockIcon(context, parameters, x, y) {
```

Comment:
Suggest to pass the concrete drawing parameters instead of the parameters object. It seem you only use `parameters.iconSize` and `strokeColor`. These kind of helper functions should be fairly generic, and tying it into the parameters object creates a tighter coupling than needed.

Reply:
thanks. done.